packages/firestore/src/model/document.ts

```diff
@@ -68,7 +81,7 @@ export class Document {
     );
   }
 
-  static compareByKey(d1: MaybeDocument, d2: MaybeDocument): number {
+  static compareByKey(d1: Document, d2: Document): number {
```

Comment:
Why shouldn't this accept `MaybeDocument`?

Reply:
I moved this method up to the base class and changed it to accept MaybeDocument. I just thought it was odd that it accepts MaybeDocument, but is part of a subclass.